public/js/analytics/averageCmc.js

```diff
@@ -0,0 +1,76 @@
+function GetColorCat(colors) {
+  if (colors.length === 0) {
+    return 'Colorless';
+  }
+  if (colors.length > 1) {
+    return 'Multi';
+  }
+  if (colors.length === 1) {
+    switch (colors[0]) {
+      case 'W':
+        return 'White';
+      case 'U':
+        return 'Blue';
+      case 'B':
+        return 'Black';
+      case 'R':
+        return 'Red';
+      case 'G':
+        return 'Green';
+      case 'C':
+      default:
+        return 'Colorless';
+    }
+  }
+}
+
+onmessage = (e) => {
+  if (!e) return;
+  const cards = e.data;
+
+  const ColorCounts = {
+    White: { label: '{w}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Blue: { label: '{u}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Black: { label: '{b}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Red: { label: '{r}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Green: { label: '{g}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Colorless: { label: '{c}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Multi: { label: '{m}', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+    Total: { label: 'Total', asfan: 0, count: 0, totalAsfan: 0, totalCount: 0 },
+  };
+
+  for (const card of cards) {
+    const asfan = card.asfan || 15 / cards.length;
+    const colorCat = GetColorCat(card.colors || card.details.color_identity);
+    const cmc = card.cmc || card.details.cmc;
```

Comment:
This won't work if the CMC is overridden to zero. See suggestion.
```suggestion
    const cmc = card.cmc !== undefined ? card.cmc : card.details.cmc;
```